### ISSUE TITLE
[Wish] Send enemy expeditions home

### ISSUE THREAD

**Issue by Spikeone** [Return-To-The-Roots/s25client#185]
If you land on a new island you can just let your ship be there forever. That way if an other player tries to overtake this island by building a harbor, you still have your expedition there and can immediatly attack the new harbor. That way I consider it more like a design fault than an addon as we changed the normal seafaring system and in the original game this couldn't happen (Because AI never used seafaring).

**Comment by Flamefire** [comment]
You cannot attack from an expedition can you? So why is this a problem? Because the other player can see you? If yes Wer had that earlier and only solution would be to include soldiers in an expedition or something like that

**Comment by Spikeone** [comment]
Or to send idle expeditions home. That'd also make a bit sense if you see it like that: ships can travel 300 nodes, the reason may be supplies, if an expedition idles those "supplies" also get used and after a while the expedition heads home.

An other easy solution is sending all expeditions home if the harbor spot they currently landing is already occupied (either an otger harbor or a player reached it with a military building).

**Comment by Flamefire** [comment]
I like the idle part. That is reasonable and easy. After how many gf would you say?
Occupation might be too hard because you basically abort the mission if a player is detected there. So idle time is better

**Comment by Spikeone** [comment]
Should be slightly shorter than it takes a harbor to finish. Problem I just imagined is: you sent a expedition, forgot is as you may had to fight or something and then your expedition is back and you have to send it again. That's the only reason why aborting expeditions if target spot is taken (abort on arrive not on taken) might be better for the gameplay itself.

**Comment by Flamefire** [comment]
Abort on arrive or only if the harbor spot gets taken? Your sentence sounds confusing ;-)

**Comment by Spikeone** [comment]
Abort on arrive. I think that'd be the besz way but till it has been tested nobod knows.

**Comment by Flamefire** [comment]
That sounds like a bad idea. Imagine the use case:
Send an expedition to an unknown harbor which takes some time for preparation and shipping. Then you accidentally discover that there is enemy territory there so your expedition gets send back and cancelled. Gonna be annoying and would even make it impossible to go through various locations and get to islands behind enemy islands,.

Abort on discover would be better: You can land where the enemy is (it already is there with buildings etc. so no need to send it back to avoid spy expeditions) but when an enemy takes that point while your expedition is there it is send away. Similar to carriers beeing send away by attacking soldiers.

**Comment by Spikeone** [comment]
Can't agree with that completely.

You can never take islands behind enemys when you need to land on a specific, hostile harbor spot. It is only possible until that spot is taken, if you sent an expedition before you were just lucky at that point.

But I do somehow agree that it is (at least should) be only be a problem if the enemy just finished its harbor. But nobody could measure that exactly. Maybe expeditions should never scout the harbor spot itself? That'd also fix the problem. As the spot itself isn't useful at all but the surrounding, which is discovered, is.

**Comment by Flamefire** [comment]
You could. You are south the enemy in the middle and north of that an unoccupied island. Currently sending the expedition up may end up on the enemy's harbor and then you can send it further up.

**Comment by Spikeone** [comment]
Yes sure, as long as you sent the expedition before the spot was taken, I'd call that rather lucky than normal.

So still maybe best, non breaking or strange behaving solution then is, that an expedition should never scout the harbor spot, maybe not even a military building at all?

**Comment by Flamefire** [comment]
If the ship can't scout the harbor spot it landed in it would be even worse. E. G. You won't be able to see why you can't found a colony there

**Comment by Spikeone** [comment]
Uhm, you see hostile borders and hostile roads perhaps or even buildings. It's quite obvious why you can't land there isn't it?

If it is blocked by an object, show it. Just don't show the harbor or any Military building. There is no way defeating those "lookout" towers.

**Comment by Flamefire** [comment]
Then whats the need to hide the harbor? The initial problem you wanted solved is that you can place a ship there to spy on when the enemy founds a harbor there and attack it right away. If you can't see the founded harbor then you will see nothing at all because there won't be anything until the harbor is finished.

So IMO the only viable solution would be to send the expedition home when the point GETS captured by an enemy. We could also skip enemy points when sending the ship to a destination, but I think this is to complicated to implement due to the many changes required to existing code.

**Comment by Spikeone** [comment]
None at all, just as you wanted to keep a way that an expedition that has been sent before the point was taken could continue. And you can see something: a flag.

But I also agree that sending expeditions home if their target spot / current landing spot is taken would be the best way for multiplayer.
